rust_src/src/buffers.rs

```diff
@@ -1843,4 +1847,80 @@ pub fn byte_char_debug_check(b: LispBufferRef, charpos: isize, bytepos: isize) {
     }
 }
 
+// True if B can be used as 'other-than-BUFFER' buffer.
+fn candidate_buffer(b: LispObject, buffer: LispObject) -> bool {
+    if let Some(buf) = Option::<LispBufferRef>::from(b) {
```

Comment:
This could be written as a match to be more idiomatic Rust.

```
match b.as_buffer() {
    Some(buf) => !b.eq(buffer) && buf.is_live() && !buf.is_hidden(),
    None => false
}
```

You might be able to use `match b.into()` here but I suspect the type checker won't allow it. `as_buffer` might end up being deprecated but it is ok to use it for now.

Reply:
Thanks, done.